### PR TITLE
Make httplistener test project build separately for each OS config.

### PR DESCRIPTION
**Pull request by Priya91**
cc @MattGal

### CODE REVIEW

**Comment by Priya91** [comment]
cc @MattGal @mmitche 

OSX build fails in pipeline step.. Passes locally on my osx 10.12 box and updates the runtests.sh with ulimit -n 7000 value.

```
[Pipeline] node
Still waiting to schedule task
Waiting for next available executor on osx-10.12
Running on dci-mac-build-105 in /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest
[Pipeline] {
[Pipeline] step
[WS-CLEANUP] Deleting project workspace...[WS-CLEANUP] done
[Pipeline] isUnix
[Pipeline] sh
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
Setting status of 9729cee59d90f473dd49ea79c1a15687a2fb9767 to FAILURE with url https://ci3.dot.net/job/dotnet_corefx/job/master/job/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/1303/ and message: 'Build finished. '
Using context: OSX x64 Debug Build
```

**Comment by MattGal** [comment]
@Priya91 change looks reasonable and innocuous, I think this was some sort of Jenkins issue and not your change.  @mmitche ?

**Comment by mmitche** [comment]
@Priya91 The node is going offline...probably network issues.

**Comment by Priya91** [comment]
@dotnet-bot test OSX x64 Debug Build

**Comment by Priya91** [comment]
Thanks restarting build again

**Comment by Priya91** [comment]
failed again at build-tests
```
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Build Tests)
[Pipeline] sh
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // timeout
[Pipeline] }
[Pipeline] // timestamps
[Pipeline] echo
Archiving logs in netci-archived-logs/**
[Pipeline] step
[Pipeline] echo
Error during cleanup: java.io.IOException: remote file operation failed: /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest at hudson.remoting.Channel@37034d61:JNLP4-connect connection from 131.107.19.218/131.107.19.218:54579: hudson.remoting.ChannelClosedException: channel is already closed
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
Setting status of 9729cee59d90f473dd49ea79c1a15687a2fb9767 to FAILURE with url https://ci3.dot.net/job/dotnet_corefx/job/master/job/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/1312/ and message: 'Build finished. '
Using context: OSX x64 Debug Build
```
restarting again,

@dotnet-bot test OSX x64 Debug Build

**Comment by mmitche** [comment]
@Priya91 This is sort of odd behavior for this CI failure.  Is the test messing around with the network at all?

**Comment by Priya91** [comment]
> @Priya91 This is sort of odd behavior for this CI failure. Is the test messing around with the network at all?

The tests in this project open a lot of sockets on localhost NIC. But, the build hasn't even gotten to the point of running the tests, it seems to be in building test phase.

File: src/System.Net.HttpListener/tests/Configurations.props [hunk 1]
```diff
@@ -2,7 +2,9 @@
 <Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
     <BuildConfigurations>
-      netstandard;
```
Comment (caesar1995) [inline src/System.Net.HttpListener/tests/Configurations.props]: If we remove the netstandard, and only add the netcoreapp-* for build configurations, then we are not able to run tests for UAP and ILC. 

@davidsh @DavidGoll

**Comment by davidsh** [comment]
@Priya91 What was the purpose of this PR?  In general, all test projects should just run against 'netstandard' configuration.

This PR appears to have broken tests since @caesar1995 says that the tests don't work on UAP and UAPAOT mode now.

**Comment by davidsh** [comment]
>This PR appears to have broken tests since @caesar1995 says that the tests don't work on UAP and UAPAOT mode now.

Although, I think, perhaps, the fix for this is not neccessarily to put back the 'netstandard' (if that was causing other issues?), but we need to expliticlty added UAP to the configuration.props file.  That should result in UAP and UAPAOT tests working again.

@safern PTAL.

**Comment by Priya91** [comment]
If this doesn't test for uap, uapaot profile, then there's a separate build moniker for that? @weshaggard @mellinoe can you help with this. Thanks!

**Comment by davidsh** [comment]
Yes, you need this:

```c#
  <PropertyGroup>
    <BuildConfigurations>
      uap-Windows_NT;
    </BuildConfigurations>
  </PropertyGroup>
</Project>
```
 
in the configuration.props file for the tests, i.e. add the "uap-Windows_NT" line.

**Comment by davidsh** [comment]
@caesar1995 Please submit a new PR to add the "uap-Windows_NT" line to the configuration.props for the tests.

**Comment by caesar1995** [comment]
Ok, I will change the configuration.props file.

**Comment by safern** [comment]
We want to test against netstandard as much as possible as the build configuration as that is the purpose of netstandard, having one implementation and multiple different runtimes with that implementation. So if the point of this PR was to have the property `TargetsOSX` set to true, you could also achieve that with a netstandard configuration by doing:

```xml
<PropertyGroup>
  <BuildConfigurations>
     netstandard-Windows_NT;
     netstandard-OSX;
     netstandard-Unix;
  <BuildConfigurations>
</PropertyGroup>
```

Here is another place where we are doing it: https://github.com/dotnet/corefx/blob/937d8b7be4915c8d2374963feb8ad580cf8b506f/src/System.Diagnostics.Process/tests/Configurations.props#L6

From what I see in this PR's .csproj nothing was modified to use `TargetGroup` expecting it to be another value different than `netstandard` so there is no need to change this configurations to be specific to netcoreapp and uap. 

So I would change the configurations.props to be as I suggest above as it is better to build the tests as netstandard when all the surface area that is being tested is part of it.

@weshaggard please correct me if I'm wrong.

**Comment by davidsh** [comment]
We already did a PR #22573 in order to fix the UAP tests which stopped running due to this PR.

Its sounds like the real, correct, fix to all of this is to put back netstandard* as you indicate above. That will fix the original OSX issue that @Priya91 had.  It will also let UAP tests run correctly.

cc: @caesar1995

**Comment by caesar1995** [comment]
Should we open an issue and change test project configuration.props for all System.Net.* contracts?

**Comment by davidsh** [comment]
>Should we open an issue and change test project configuration.props for all System.Net.* contracts?

No.  We already use 'netstandard' in most test projects.  The only one that got messed up is HttpListener.  So, just submit a PR to fix that.  Please add @Priya91 @safern and ncldev as reviewers.

**Pull request merged**